src/components/LightningConnectWallet.vue

```diff
@@ -0,0 +1,122 @@
+        disabled
+        >Select your wallet</b-form-select-option
+      >
+      <b-form-select-option
+        :value="{ walletName: 'Zap Desktop', type: 'grpc', network: 'tor' }"
+        >Zap Desktop</b-form-select-option
+      >
+      <b-form-select-option
+        :value="{ walletName: 'Zap Mobile', type: 'rest', network: 'tor' }"
+        >Zap Mobile</b-form-select-option
+      >
+      <b-form-select-option
+        :value="{ walletName: 'Zeus', type: 'rest', network: 'local' }"
+        >Zeus</b-form-select-option
+      >
+      <b-form-select-option-group label="Other">
+        <b-form-select-option
+          :value="{
+            walletName: 'any other wallet that uses lndconnect gRPC',
+            type: 'grpc',
+            network: 'local'
+          }"
+          >lndconnect gRPC</b-form-select-option
+        >
+        <b-form-select-option
+          :value="{
+            walletName: 'any other wallet that uses lndconnect gRPC (Tor)',
+            type: 'grpc',
+            network: 'tor'
+          }"
+          >lndconnect gRPC (Tor)</b-form-select-option
+        >
+        <b-form-select-option
+          :value="{
+            walletName: 'any other wallet that uses lndconnect REST',
+            type: 'rest',
+            network: 'local'
+          }"
+          >lndconnect REST</b-form-select-option
+        >
+        <b-form-select-option
+          :value="{
+            walletName: 'any other wallet that uses lndconnect REST (Tor)',
+            type: 'rest',
+            network: 'tor'
+          }"
+          >lndconnect REST (Tor)</b-form-select-option
+        >
+      </b-form-select-option-group>
+    </b-form-select>
+    <div
+      class="d-lg-flex align-items-center"
+      v-if="lndConnectUrl && selectedWallet.type"
+    >
+      <!-- Pubkey QR Code -->
+      <qr-code
+        :value="lndConnectUrl"
+        :size="200"
+        level="M"
+        class="qr-image mb-3 mb-lg-0"
+        showLogo
+      ></qr-code>
+      <div class="w-100 align-self-center ml-lg-3">
+        <p class="text-center text-lg-left">
+          Connect {{ selectedWallet.walletName }} to your Umbrel using the
+          following url
+        </p>
+        <input-copy class="mb-2" size="sm" :value="lndConnectUrl"></input-copy>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapState } from "vuex";
+
+import QrCode from "@/components/Utility/QrCode";
+import InputCopy from "@/components/Utility/InputCopy";
+
+export default {
+  data() {
+    return {
+      selectedWallet: {
+        walletName: "Select your wallet",
+        type: null,
+        network: null
+      }
+    };
+  },
+  computed: {
+    ...mapState({
+      lndConnectUrl: state => state.lightning.lndConnectUrl
+    })
+  },
+  methods: {},
+  watch: {
+    selectedWallet: async function(wallet) {
+      this.$store.dispatch("lightning/getLndConnectUrl", {
```

Comment:
This is triggering a network round trip each time the user changes the select, waiting for the round trip to complete, then updating the result in the UI.

Over Tor this is quite noticeable.

It's also confusing because the text content is updated immediately but the connection string and QR wait for the network. So if you change between wallets over Tor there's a period of a few seconds where the text is saying one thing but the connection string is for the previously selected option.

![umbrel-lag](https://user-images.githubusercontent.com/2123375/94255090-e81efc00-ff51-11ea-9c66-29edddcf95c9.gif)

I think it would be better to request all the required information on Lightning page load and then instantly show the result to the user when they select it instead of waiting for extra network requests to complete.

Reply:
Very good point. I like how you did it on the Bitcoin page. Will update this to the same pattern.

Reply:
Ah wait, I'm not sure if this would be the right approach as we'd have to make 4 different API calls right on the page load, then store all 4 types of endpoints in the global store. Here's the current API call which is made only when the input is changed, depending upon the wallet type:

https://github.com/getumbrel/umbrel-dashboard/blob/1a6ad37735752f4525378222123d140fe504ca36/src/store/modules/lightning.js#L437

Usually, it's much better to create a loading view in between the API calls (so for example during the lag that's there in your gif, it could show a loading bar or something), but that seems like an overkill for now, especially when during the intended use the user would only be selecting one wallet at a time instead of trying every option.

Reply:
>Ah wait, I'm not sure if this would be the right approach as we'd have to make 4 different API calls right on the page load, then store all 4 types of endpoints in the global store.

I'd say that's better than the current approach. The 4 requests can be async so they don't block anything.

We could probably quite easily just squash it down to a single request though. Does the `network` param just swap out the .onion for umbrel.local? If so we can just do that client side anyway.

It could be a single request to `/v1/system/lndconnect-url` that returns an object like:

```json
{
  rest: "blah.onion?abc"
  rpc: "blah.onion?def"
}
```